memory_addr/src/range.rs

```diff
@@ -98,7 +92,11 @@ where
     /// ```
     #[inline]
     pub fn size(self) -> usize {
-        usize!(self.end) - usize!(self.start)
+        if self.is_empty() {
```

Comment:
Why is there an `if`?

Reply:
To handle empty ranges with `start > end`. I think we should make it clear what it means and what the behavior these methods should have when `start > end`. Maybe just reject them with a checked `new` and write a `new_unchecked`?